### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by gealsanchez**
Add the following validations:
For the User model:
Name must not be blank.
PostsCounter must be an integer greater than or equal to zero.
For the Post model:
Title must not be blank.
Title must not exceed 250 characters.
CommentsCounter must be an integer greater than or equal to zero.
LikesCounter must be an integer greater than or equal to zero.
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  it 'posts counter should be greater than or equal to 0' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'should return the 3 most recent posts for a given user' do
+    expect(subject.recent_posts.count).to eq(0)
+  end
```
Comment (TadesseDev) [inline spec/user_spec.rb]: - [Optional] Dear @gealsanchez, Though you have a test for the `recent_posts` currently it's not efficient as you did not have any posts for the user. A better analogy for this I believe is to add 4-5 posts and test it returns the most recent/(3) posts.

File: spec/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+require_relative 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  subject do
+    Post.new(title: 'first_post', text: 'This is my first post', comments_counter: 1, likes_counter: 1, author_id: 1)
+  end
```
Comment (TadesseDev) [inline spec/post_spec.rb]: - [ ] Dear @gealsanchez, according to the primary requirements for the project, any custom methods in the models are expected to have a corresponding unit test. As part of that, we need to add a test for the `update_posts_counter` method inside your `Post` model.

File: spec/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  it 'likes counter should be integer' do
+    subject.likes_counter = 1.1
+    expect(subject).to_not be_valid
+  end
+
+  it 'should return the 3 most recent comments for a given post' do
+    expect(subject.recent_comments.count).to eq(0)
+  end
```
Comment (TadesseDev) [inline spec/post_spec.rb]: - [Optional] Dear @gealsanchez, Though you have a test for the `recent_comments` currently it's not efficient as you did not have any comment for the post. A better analogy for this I believe is to add 6-7 comments and test it returns the most recent/(5) comments.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @gealsanchez 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [Optional] Please try to separate your code into different levels of commits. Remember your commits are going to help you look back and debug your code. It's a good idea to commit after every visible change.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @gealsanchez,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**